src/sort.rs

```diff
@@ -45,8 +51,14 @@ pub fn sort_toml(
         if toml.as_table().contains_key(heading) {
             if let Item::Table(table) = &mut toml[heading] {
                 if table.contains_key(key) {
-                    if let Item::Value(Value::Array(arr)) = &mut table[key] {
-                        sort_array(arr);
+                    match &mut table[key] {
+                        Item::Value(Value::Array(arr)) => {
+                            sort_array(arr);
+                        }
+                        Item::Table(table) => {
+                            sort_table(table, group);
+                        }
+                        _ => unreachable!("{}.{} must be value or table", heading, key),
```

Comment:
Should we really panic here? Can't we just ignore this case and/or print a warning?

I think it would be a better UX if we were to just ignore input that we don't understand rather than panic.

Reply:
Could print to stderr using `eprintln!`? Seems like it should be good enough.

Reply:
We can yeah. I'd think that if we wanted to add logging, we should do it properly via `log` or `tracing` but that seems out of scope for this PR.